e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,342 @@
+)
+
+const (
+	yamlDir = "../testdata"
+)
+
+func TestNoDefaultFieldsInNamespaceConfig(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2, ntopts.SkipMultiRepo)
+	expectedName := "hello-world"
+	expectedTime := metav1.Time{}
+
+	nt.T.Log("Add a deployment")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/deployment-helloworld.yaml", yamlDir), "acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Adding deployment")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Check that the deployment was created")
+	if err := nt.Validate("hello-world", "dir", &appsv1.Deployment{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that specified field name is present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{},
+		func(o client.Object) error {
+			raw := o.(*v1.NamespaceConfig).Spec.Resources[0].Versions[0].Objects[0].Raw
+			dep := &appsv1.Deployment{}
+			if err := json.Unmarshal(raw, &dep); err != nil {
+				return errors.Errorf("Failed to get raw JSON, %s", err)
+			}
+			actualName := dep.GetName()
+			if actualName != expectedName {
+				return errors.Errorf("Expected name: %s, got: %s", expectedName, actualName)
+			}
+			return nil
+		}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Check that unspecified field creationTimestamp is not present")
+	if err := nt.Validate("dir", "", &v1.NamespaceConfig{},
+		func(o client.Object) error {
+			raw := o.(*v1.NamespaceConfig).Spec.Resources[0].Versions[0].Objects[0].Raw
+			dep := &appsv1.Deployment{}
+			if err := json.Unmarshal(raw, &dep); err != nil {
+				return errors.Errorf("Failed to get raw JSON, %s", err)
+			}
+			actualTime := dep.GetCreationTimestamp()
+			if actualTime != expectedTime {
+				return errors.Errorf("Expected time: %s, got: %s", expectedTime, actualTime)
+			}
+			return nil
+		}); err != nil {
+		nt.T.Fatal("NamespaceConfig has default field creationTimestamp which was not specified")
+	}
+}
+
+func TestRecreateSyncDeletedByUser(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation2, ntopts.SkipMultiRepo)
+
+	nt.T.Log("Add a deployment to a directory")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/dir-namespace.yaml", yamlDir), "acme/namespaces/dir/namespace.yaml")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/deployment-helloworld.yaml", yamlDir), "acme/namespaces/dir/deployment.yaml")
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add a deployment to a directory")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Ensure that the system created a sync for the deployment")
+	if err := nt.Validate("deployment.apps", "default", &v1.Sync{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Force-delete the sync from the cluster")
+	sync := &v1.Sync{}
+	if err := nt.Get("deployment.apps", "default", sync); err != nil {
```

Comment:
You don't need to get the object before deleting it.

```
nt.Delete(fake.SyncObject(kinds.Deployment().GroupKind(), core.Namespace("default")))
```

Reply:
thanks for the suggestion! I didn't think of doing it this way